org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/Junit4Fragment2.xtend

```diff
@@ -27,11 +27,19 @@ class Junit4Fragment2 extends AbstractStubGeneratingFragment {
 			"org.eclipse.xtext.testing"
 	}
 	
+	def protected getXbaseTestingPackage() {
+		if (useDeprecatedClasses)
+			"org.eclipse.xtext.xbase.junit"
+		else
+			"org.eclipse.xtext.xbase.testing"
+	}
+	
 	override generate() {
 		if (projectConfig.runtimeTest.manifest !== null) {
 			projectConfig.runtimeTest.manifest => [
 				requiredBundles.addAll(
 					testingPackage,
+					xbaseTestingPackage,
```

Comment:
Yes, but the difference is that this code will add the dependency even if the user has removed it before, so there is no way to use the fragment without having that dependency. The project generator of the wizard, on the other hand, is a one-time generator.